addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/connector/RFXComStreamReader.java

```diff
@@ -37,89 +36,39 @@ public RFXComStreamReader(RFXComBaseConnector connector, InputStream in) {
     }
 
     @Override
-    public void interrupt() {
-        interrupted = true;
-        super.interrupt();
-        try {
-            in.close();
-        } catch (IOException ignore) {
-            // quietly close
-        }
-    }
-
-    @Override
     public void run() {
-        final int dataBufferMaxLen = Byte.MAX_VALUE;
-
-        byte[] dataBuffer = new byte[dataBufferMaxLen];
-
-        int msgLen = 0;
-        int index = 0;
-        boolean startFound = false;
-
         logger.debug("Data listener started");
 
-        try {
-
-            byte[] tmpData = new byte[20];
-            int len = -1;
-
-            while (!interrupted) {
-
-                if ((len = in.read(tmpData)) > 0) {
-
-                    byte[] logData = Arrays.copyOf(tmpData, len);
-                    logger.trace("Received data (len={}): {}", len, DatatypeConverter.printHexBinary(logData));
+        while (!Thread.interrupted()) {
+            try {
+                byte[] buf = new byte[Byte.MAX_VALUE];
 
-                    for (int i = 0; i < len; i++) {
+                if (in.read(buf, 0, 1) > 0 && buf[0] > 0) {
+                    int i = 1;
+                    int intCount = 0;
 
-                        if (index > dataBufferMaxLen) {
-                            // too many bytes received, try to find new
-                            // start
-                            startFound = false;
-                        }
-
-                        if (!startFound && tmpData[i] > 0) {
-
-                            startFound = true;
-                            index = 0;
-                            dataBuffer[index++] = tmpData[i];
-                            msgLen = tmpData[i] + 1;
-
-                        } else if (startFound) {
-
-                            dataBuffer[index++] = tmpData[i];
-
-                            if (index == msgLen) {
-                                sendMessage(dataBuffer, msgLen);
+                    while (i <= buf[0]) {
+                        int n = in.read(buf, i, buf[0] - i + 1);
 
-                                // find new start
-                                startFound = false;
-                            }
+                        if (n > 0) {
+                            i += n;
+                        } else if (intCount++ == 4) {
+                            throw new IOException("Timeout during packet read");
```

Comment:
Yes we should. We should not expect to see _any_ read timeouts in the middle of a packet really. But if they _do_ occur it means the RFXCOM is not doing what it should be doing and we cannot just ignore it. That would leave the bridge ONLINE but simply not giving us the input it should which means your lights stop working. The least we can do is throw an error up the stack to trigger a reconnect and reset.